### DISCUSSION TITLE
SQL commands not working

### DISCUSSION THREAD

**Discussion by AJPF1** [dbeaver/dbeaver#21388] [Q&A]
I am relatively new to SQL so I have to frequently look up how to do stuff. 
What seems like most of time, though, my internet search returns what could be useful SQL code but, when I try it in dBeaver, it doesn't work. 
The latest one with which I have a problem is JSON_SIZE - I just get a "Unknown function" message. 
Does dBeaver have a some kind of restriction on which SQL commands work and which won't?

**Comment by LonwoLonwo** [comment]
Hello @AJPF1 

Usually, it is database restrictions, not DBeaver's.

Could you provide an example for the last error? With the full error log.

**Comment by AJPF1** [comment]
Hello, 

the code I am trying to run is: 

select *, 
       JSON_SIZE(json, '$.products') as size
from [TABLE NAME]
where JSON_VALUE(json, '$.domain') = '[DOMAIN NAME]' and 
      JSON_VALUE(json, '$.products') is not null
order by timestamp desc 

and the error message is this: 

SQL Error [46] [07000]: Code: 46. DB::Exception: Unknown function JSON_SIZE: While processing SELECT timestamp, hash, json, JSON_SIZE(json, '$.products') AS size FROM [TABLE NAME] WHERE (JSON_VALUE(json, '$.domain') = '[DOMAIN NAME]') AND (JSON_VALUE(json, '$.products') IS NOT NULL) ORDER BY timestamp DESC. (UNKNOWN_FUNCTION) (version 22.2.2.1)
, server ClickHouseNode [uri=http://xx.xx.xx.xx:xxxx/default, options={use_server_time_zone=false,use_time_zone=false}]@-xxxxxxxxxx

**Reply by LonwoLonwo** [reply]
I guess you're trying to use the SQL function from the other database for Clickhouse. It will not work. Each database has support for different SQL functions and keywords. 
These are JSON functions for CLickhouse: https://clickhouse.com/docs/en/sql-reference/functions/json-functions
